shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/viewModel/FavoritesViewModel.kt

```diff
@@ -0,0 +1,186 @@
+        var staticRouteCardData: List<RouteCardData>? by remember { mutableStateOf(null) }
+
+        var active: Boolean by remember { mutableStateOf(true) }
+        var alerts: AlertsStreamDataResponse? by remember { mutableStateOf(null) }
+        var location: Position? by remember { mutableStateOf(null) }
+        var now: Instant by remember { mutableStateOf(Clock.System.now()) }
+
+        val globalData = getGlobalData("FavoritesViewModel.getGlobalData")
+        val stopIds =
+            remember(favorites, globalData) {
+                val stops = favorites?.mapNotNull { globalData?.getStop(it.stop) }
+                stops?.flatMap { stop ->
+                    stop.childStopIds.filter { globalData?.stops?.containsKey(it) ?: false } +
+                        stop.id
+                }
+            }
+        val schedules = getSchedules(stopIds, "FavoritesViewModel.getSchedules")
+        val predictions = subscribeToPredictions(stopIds, active)
+
+        LaunchedEffect(Unit) { favorites = favoritesUsecases.getRouteStopDirectionFavorites() }
+
+        LaunchedEffect(Unit) {
+            events.collect { event ->
+                when (event) {
+                    Event.ReloadFavorites ->
+                        favorites = favoritesUsecases.getRouteStopDirectionFavorites()
+                    is Event.SetActive -> {
+                        active = event.active
+                        event.isReturningFromBackground?.let { isReturningFromBackground = it }
+                    }
+                    is Event.SetAlerts -> alerts = event.alerts
+                    is Event.SetLocation -> location = event.location
+                    is Event.SetNow -> now = event.now
+                }
+            }
+        }
+
+        LaunchedEffect(predictions) {
+            if (predictions != null) {
+                isReturningFromBackground = false
+            }
+        }
+
+        LaunchedEffect(
+            stopIds,
+            globalData,
+            location,
+            schedules,
+            predictions,
+            alerts,
+            now,
+            favorites,
+        ) {
+            if (stopIds == null || globalData == null || location == null) {
+                routeCardData = null
+            } else if (stopIds.isEmpty()) {
+                routeCardData = emptyList()
+            } else {
+                val loadedRouteCardData =
+                    RouteCardData.routeCardsForStopList(
+                        stopIds,
+                        globalData,
+                        location,
+                        schedules,
+                        predictions,
+                        alerts,
+                        now,
+                        emptySet(),
+                        RouteCardData.Context.Favorites,
+                    )
+                routeCardData = filterRouteAndDirection(loadedRouteCardData, globalData, favorites)
```

Comment:
thought(non-blocking): This would be a further refactor so understandably not in scope for this PR, but wondering if here and for the staticRouteCardData it would be worth passing the favorites as a parameter so we don't have to filter after the fact.